pkg/server/chaosd/disk.go

```diff
@@ -35,87 +36,135 @@ type diskAttack struct{}
 var DiskAttack AttackType = diskAttack{}
 
 const DDWritePayloadCommand = "dd if=/dev/zero of=%s bs=%s count=%s oflag=dsync"
-const DDReadPayloadCommand = "dd if=%s of=/dev/null bs=%s count=%s iflag=dsync,direct,fullblock"
+const DDReadPayloadCommand = "dd if=%s of=/dev/null bs=%s count=%s iflag=dsync,fullblock,nocache"
 
 func (disk diskAttack) Attack(options core.AttackConfig, env Environment) (err error) {
-	attack := options.(*core.DiskCommand)
+	attack := options.(*core.DiskOption)
 
 	if options.String() == core.DiskFillAction {
-		return disk.attackDiskFill(attack)
+		return disk.diskFill(attack)
 	}
-	return disk.attackDiskPayload(attack)
+	return disk.diskPayload(attack)
 }
 
-func (diskAttack) attackDiskPayload(payload *core.DiskCommand) error {
+// diskPayload will execute a dd command (DDWritePayloadCommand or DDReadPayloadCommand)
+// to add a write or read payload.
+func (diskAttack) diskPayload(payload *core.DiskOption) error {
+	var cmdFormat string
 	switch payload.Action {
```

Comment:
Can we abstract this logic to a single function? such as `initPlayloadPath` or others.

Reply:
OK, I agree with you.

Reply:
Done. But I could not abstract the logic that temp file of wirte payload should be ```defer remove``` in the diskPayload func.